packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx

```diff
@@ -362,17 +369,17 @@ export function MultipleConnectionSidebar({
 
   const onNewConnectionConnect = useCallback(
     (connectionInfo: ConnectionInfo) => {
-      void connect({
-        ...cloneDeep(connectionInfo),
-      }).then(() => setIsConnectionFormOpen(false));
+      setIsConnectionFormOpen(false);
+      void _onConnect(connectionInfo).then(() => {
```

Comment:
We weren't using all the same code for new connections as when connecting to existing saved connections which meant that errors weren't caught and the toast didn't appear.